src/components/MultiSelect/MultiSelect.stories.tsx

```diff
@@ -0,0 +1,548 @@
+              Maecenas sed diam eget risus varius blandit sit amet non magna.
+              Cum sociis natoque penatibus et magnis dis parturient montes,
+              nascetur ridiculus mus. Aenean lacinia bibendum nulla sed
+              consectetur. Donec sed odio dui.
+            </Text>
+          </Stack>
+        </div>
+      </Stack>
+    </>
+  ),
+};
+
+export const withDefaultOpenStateExample: Story = {
+  render: () => (
+    <MultiSelectStory
+      id="multi-select-id"
+      isDefaultOpen={true}
+      items={withChildrenItems}
+    />
+  ),
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=43593%3A24611",
+    },
+    jest: ["MultiSelect.test.tsx"],
+  },
+};
+
+const MultiSelectWithControlsStory = (args) => {
+  // Example with custom hook useMultiSelect.
+  const { onChange, onMixedStateChange, onClear, selectedItems } =
+    useMultiSelect();
+
+  // Hack to get storybook's action tab to log state change when selectedItems state changes.
+  const [actionName, setActionName] = useState("");
+
+  useEffect(() => {
+    if (Object.keys(selectedItems).length !== 0) {
+      action(actionName)(selectedItems);
+    }
+    if (actionName === "onClear") {
+      action(actionName)(selectedItems);
+    }
+  }, [actionName, selectedItems]);
+  const multiSelectId = args.id;
+
+  return (
+    <MultiSelect
+      {...args}
+      helperText="multi-select-helper-text"
+      defaultItemsVisible={5}
+      selectedItems={selectedItems}
+      onChange={(e) => {
+        onChange(e.target.id, multiSelectId);
+        setActionName("onChange");
+      }}
+      onMixedStateChange={(e) => {
+        onMixedStateChange(e.target.id, multiSelectId, args.items);
+        setActionName("onMixedStateChange");
+      }}
+      onClear={() => {
+        onClear(multiSelectId);
+        setActionName("onClear");
+      }}
+    />
+  );
+};
+
+const MultiSelectStory = ({
+  id,
+  isBlockElement = false,
+  isSearchable = false,
+  isDefaultOpen,
+  items,
+  listOverflow = "scroll",
+  width = "full",
+}) => {
+  // Example with custom hook useMultiSelect.
+  const { onChange, onMixedStateChange, onClear, selectedItems } =
+    useMultiSelect();
+
+  // Hack to get storybook's action tab to log state change when selectedItems state changes.
+  const [actionName, setActionName] = useState("");
+
+  useEffect(() => {
+    if (Object.keys(selectedItems).length !== 0) {
+      action(actionName)(selectedItems);
+    }
+    if (actionName === "onClear") {
+      action(actionName)(selectedItems);
+    }
+  }, [actionName, selectedItems]);
+
+  return (
+    <MultiSelect
+      buttonText="MultiSelect"
+      defaultItemsVisible={5}
+      helperText="multi-select-helper-text"
+      id={id}
+      isBlockElement={isBlockElement}
+      isDefaultOpen={isDefaultOpen}
+      isSearchable={isSearchable}
+      items={items}
+      listOverflow={listOverflow}
```

Comment:
@EdwinGuzman I added this prop to allow for better control in the stories, but there is a type issue and I am not sure how to resolve.  PLease refer to the errors shown at the top of this file.